### PR TITLE
Add initial support for Flatcar Linux

### PR DESCRIPTION
**Pull request by jkroepke**
*Issue #, if available:*

*Description of changes:*

Add support for Flatcar Linux. https://www.flatcar.org/

Since Flatcar is an immutable OS system, I mirror the Bottlerocket logic around the updater plugin.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by jkroepke** [comment]
Hold this, please. I forgot the package manager handling.

File: RELEASENOTES.md [hunk 1]
```diff
@@ -1,5 +1,6 @@
 Latest
 ===============
+- Added platform detection for Flatcar Linux
```
Comment (Thor-Bjorgvinsson) [inline RELEASENOTES.md]: Please remove this line since we need to update release notes internally

**Comment by Thor-Bjorgvinsson** [comment]
Thank you for posting this CR! couple of questions:

1. With this current change, does the agent detect the correct OS Platform? The agents send the detected OS Platform to health and can be seen either in the debug logs or from describe-instance-information API
2. Does Flatcar need to support agent update? Dependency for agent update is that the agent is run as a service via systemd which I believe is not in environments where agent is running as a service, (see [here](https://github.com/aws/amazon-ssm-agent/commit/12ee5859a99e194b1eeb90972746883973efaaf2#diff-126ddd02c2ed6571ca773df88b103ea83435dfabe133df9b65dc36c7ee4472e0) how bottle rocket added a precondition to determine if agent update should be supported)

Once the PR is final, please rebase with mainline and squash commits into one

**Comment by jkroepke** [comment]
Hi @Thor-Bjorgvinsson 

> With this current change, does the agent detect the correct OS Platform?

Yes, I also add a test which successfully detect the OS Platform based on the /etc/os-release files.

>  Dependency for agent update is that the agent is run as a service via systemd which I believe is not in environments where agent is running as a service

On Flatcar, amazon-ssm-agent runs as systemd server and the binaries are mutable. But there is no pkg manager.

Here is the [unit file](https://github.com/flatcar-linux/coreos-overlay/blob/ab16e18fb452c5945a669e520e09a4f77a6ed066/coreos-base/oem-ec2-compat/files/base/base-ec2.ign#L14):

```ini
[Unit]
Description=amazon-ssm-agent
After=network-online.target

[Service]
Type=simple
WorkingDirectory=/usr/share/oem
ExecStart=/usr/share/oem/bin/amazon-ssm-agent
KillMode=process
Restart=on-failure
RestartForceExitStatus=SIGPIPE
RestartSec=15min

[Install]
WantedBy=multi-user.target
```

**Comment by jkroepke** [comment]
@Thor-Bjorgvinsson @aguman-aws Can you please take a look here again?

**Comment by jkroepke** [comment]
@rhaoran-aws @Thor-Bjorgvinsson @aguman-aws I would appreciate an review here.

**Comment by jkroepke** [comment]
@cjinaws could you take a look here?

**Comment by jkroepke** [comment]
@gianniLesl @cjinaws @rhaoran-aws @Thor-Bjorgvinsson @aguman-aws I would appreciate a review here.

**Comment by gianniLesl** [comment]
Change looks good, will test internally and pull in for release

**Comment by jkroepke** [comment]
@gianniLesl Do you already have some feedback?

**Comment by gianniLesl** [comment]
Hi, this is in the pipeline for release later this month

**Comment by jkroepke** [comment]
@gianniLesl Do you have any information, when the release is availible?

**Comment by gianniLesl** [comment]
This changed was pushed with [v3.2.1542.0](https://github.com/aws/amazon-ssm-agent/releases#:~:text=VishnuKarthikRavindran-,3.2.1542.0,-ae76116)

**Pull request closed without merging**